Sources/StreamChatSwiftUI/ChatThreadList/ChatThreadList.swift

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import SwiftUI
+
+/// Stateless component for the channel list.
+/// If used directly, you should provide the thread list.
+public struct ThreadList<Factory: ViewFactory, HeaderView: View, FooterView: View>: View {
```

Comment:
Shouldn't the header and footer view be part of the view factory?

Reply:
I think I did this because the headerView and the footerView need the viewModel, and the ThreadList is supposedly stateless and does not know about the view model